Design/Workbench/Plot-Selector/design-plot-selector.md

```diff
@@ -0,0 +1,66 @@
+# Plot Selector Widget
+
+## Requirements
+
+The request from the January 2018 user meeting was to have a widget that allowed easy switching between plots, in particular for SANS and Reflectometry users. Another request from the ILL was to have a method to easily arrange the plots.
+
+For consistency the design should follow as closely as possible similar functionality in Mantid, such as the workspace selector and plotting options within the workspace selector.
+
+## Functional Design
+
+### Interface Design
+
+For consistency the plot selector widget should follow a similar design to the workspace selector. Double-clicking a plot name will bring the plot to the foreground and make it the active one.
+
+![Plot Selector Widget](plot_selector.png)
+
+**All plots created from the workbench are shown in the list.** Double-clicking a plot will bring it to the foreground. Filtering works as for the workspace selector. Close can act on one or more selected plots.
+
+![Menu Options](menu_options.png)
+
+**This figure shows the menu options for New, Arrange, Sort and Export.** New allows a choice of plot types, the same selection as for right-clicking a workspace. Arrange allows all plot windows to be cascaded or tiled. Sort allows sorting by name or last modified, ascending or descending. Export allows all selected plots to be exported to commonly used file types.
```

Comment:
I like the idea of this matching how the workspace list works. By name here would you mean the title of the plot?

The other question is how we would deal with the `last modified` - does matplotlib store this type of information?

Reply:
Yes, I think the name would be the title of the plot/figure window. Currently these are Figure 1, Figure 2 etc. but I guess we should follow something like the MantidPlot convention to give these a sensible automatic name.

As far as I know matplotlib does not store this information, though we can see if is in there. It might be tricky to keep track of, so this needs investigation to see if it is possible.

Reply:
I would be tempted to move the selection of whether to plot error bars to the following dialog when you are selecting the workspace to plot.

We don't do this with the right click menu for workspaces, but that is because we do not intend on showing any more dialogs.

Something similar to the plot advanced dialog.
![image](https://user-images.githubusercontent.com/1017173/40538245-17a06c88-6009-11e8-84da-2877cb0366b9.png)

Reply:
It might be worth considering something like this
![image](https://user-images.githubusercontent.com/1017173/40542687-0d27f158-6019-11e8-8d2e-f75d7e26545f.png)


 where the right hand side is the Advanced plot gui that we have to build anyhow, the only addition would be the workspace list (and filter for when they have lots of workspaces).

Reply:
Yes, this seems like a good idea. Instead of `New` being a menu it can just be a button to bring up such an advanced plot dialogue.

Reply:
That feels good but possibly becoming an extension now rather than part of the original scope. I think the the list, filtering etc is valuable without requiring the new dialog. If you have time then maybe that's something you could also look at after this?

Reply:
Yes, I agree with this. I think the new dialogue can be pushed into the 'nice to have' list. In any case this should be implemented after the advanced plot window is available, as this would be very much a secondary way of accessing it.